TunnelVisionLabs.ReferenceAssemblyAnnotator/WellKnownTypes+MaybeNullWhenAttributeProvidedType.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Tunnel Vision Laboratories, LLC. All Rights Reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace TunnelVisionLabs.ReferenceAssemblyAnnotator
+{
+    using System;
+    using Mono.Cecil;
+
+    internal partial class WellKnownTypes
+    {
+        private sealed class MaybeNullWhenAttributeProvidedType : ProvidedType
+        {
+            public MaybeNullWhenAttributeProvidedType()
+                : base("System.Diagnostics.CodeAnalysis", "MaybeNullWhenAttribute")
+            {
+            }
+
+            protected override TypeReference DefineAttribute(ModuleDefinition module, WellKnownTypes wellKnownTypes, CustomAttributeFactory attributeFactory)
+            {
+                var attribute = new TypeDefinition(
+                    @namespace: NamespaceName,
+                    name: TypeName,
+                    TypeAttributes.NotPublic | TypeAttributes.Sealed | TypeAttributes.BeforeFieldInit,
+                    wellKnownTypes.Module.ImportReference(wellKnownTypes.SystemAttribute));
```

Comment:
Protected helper methodss could cut down on some of the remaining repetition.

Reply:
Yeah that works nicely. Now implemented.